### PR TITLE
Dropdown: improve multiselect accessibility

### LINKED ISSUE #10400: Narrator does not announce the multiselected options of dropdown
### Environment Information

- 7.31.1: (fill this out)
- Edge, Chrome: (fill this out if relevant)

### Describe the issue:

Go to the office fabric page with dropdown examples.
Start Narrator.
Start tabbing with your keyboard till you find the example with multi-select Dropdown.
See what is announced by the Narrator - it announces the label of it but not the preselected options.

#### Actual behavior:
Go to the office fabric page with dropdown examples.
Start Narrator.
Start tabbing with your keyboard till you find the example with multi-select Dropdown.
See what is announced by the Narrator - it announces the label of it but not the preselected options.

#### Expected behavior:

I expect the Narrator to say which items are selected.

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #10400
- [X] Include a change request file using `$ yarn change`

#### Description of changes

These changes improve multiselect accessibility for Dropdown.

Before, when focused on a collapsed, multiselect dropdown, none of the selected options would be read out. Additionally, only one of the options was included in the `aria-label`. With these changes, now all of the selected options are read out and all of the selected options are included in the `aria-label`.

Before:
**Multi-select uncontrolled example, group, collapsed,** 

After:
**Multi-select uncontrolled example, Apple, Banana, Grape selected, selection contains <say-as interpret-as="spell">7</say-as> items,** 

In making these changes, I found it to be useful to separate out the logic that joins all of the selected options' text from an array into one comma separated string into another function. I reuse this to generate the `aria-label` for dropdowns. I called it `_joinSelectedOptions`, but let me know if something would sound better!

Lastly, aria-setsize is now included on multiselect dropdowns. This seems like useful info not just for single select dropdowns.

#### Focus areas to test

Ensure that selected options are read out when focusing on a collapsed dropdown.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10414)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx [hunk 1]
```diff
@@ -208,31 +207,41 @@ export class DropdownBase extends React.Component<IDropdownInternalProps, IDropd
...
+          ariaPosInSet: undefined,
+          ariaSelected: undefined
+        }
+      : multiSelect
+      ? {
+          role: 'listbox',
+          ariaActiveDescendant:
+            isOpen && selectedIndices.length === 1 && selectedIndices[0] >= 0 ? this._id + '-list' + selectedIndices[0] : optionId,
+          childRole: 'option',
+          ariaSetSize: this._sizePosCache.optionSetSize,
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: ""Apple, Banana, Grape selected, selection contains 7 items" - doesn't seem right.. wouldn't the selection have only 3 items? I think ariaSetSize should be undefined here

Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: I'm curious if there are any problems you noticed if you revert line 216-244.. wondering if these changes are necessary (naively without testing out, I would assume keeping line 210, 294 and 456-467 are good enough to fix the bug)

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: Set size seems to be the total amount of options in the dropdown. For single select dropdowns, it reads out "selection contains 7 items" as well.

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: From my testing, we don't achieve the proper readout without these attributes combined. I can double check to see if there are any extras I've added that we don't need.

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: Alright so just double checked, and everything is needed except for `ariaActiveDescendant`. Without `role`, `childRole`, and `ariaSelected`, it does not read out the selected options in a multi-select dropdown.

**Comment by KevinTCoughlin** [comment]
@joschect @cliffkoh please take another pass now that https://github.com/OfficeDev/office-ui-fabric-react/pull/10414/commits/022c1575f3cefa18f3906e15b97279c2748e2165 has been pushed.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx [hunk 2]
```diff
@@ -208,31 +207,41 @@ export class DropdownBase extends React.Component<IDropdownInternalProps, IDropd
...
-          {
-            role: 'listbox',
-            ariaActiveDescendant:
-              isOpen && selectedIndices.length === 1 && selectedIndices[0] >= 0 ? this._id + '-list' + selectedIndices[0] : optionId,
-            childRole: 'option',
-            ariaSetSize: this._sizePosCache.optionSetSize,
-            ariaPosInSet: this._sizePosCache.positionInSet(selectedIndices[0]),
-            ariaSelected: selectedIndices[0] === undefined ? undefined : true
-          };
+    const ariaAttrs = disabled
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: Nit: at this point I feel like this logic should be put into a function somewhere. It's a bit big to have in the render. Not a blocker though.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx (hunk 1 above)
Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: size is needed in single select for a different reason. Let me test it out/sync with you

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx (hunk 2 above)
Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: Good call!

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx (hunk 1 above)
Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: Hey @natalieethell, I'll need more time to close on this. I'll look into this tomorrow and touch base with you :)

**Comment by cliffkoh** [comment]
@msft-github-bot please make sure @cliffkoh approves

**Comment by natalieethell** [comment]
Chatted with Cliff and it looks like this fix is going to involve some more investigation into how Narrator works with `aria-describedby`. I'll close this PR and open another one to just add correct aria labels for multi-select dropdowns.

**Pull request closed without merging**